bitnami/pytorch/templates/deployment.yaml

```diff
@@ -0,0 +1,143 @@
+        - bash
+        - -c
+        - |
+          {{- if .Values.pytorch.entrypoint.file }}
+          python {{ .Values.pytorch.entrypoint.file }} {{ if .Values.pytorch.entrypoint.args }}{{ .Values.pytorch.entrypoint.args }}{{ end }}
+          {{- end }}
+          sleep infinity
+        env:
+        {{- if .Values.pytorch.distributed.enabled }}
+        - name: MASTER_ADDR
+          value: "127.0.0.1"
+        - name: MASTER_PORT
+          value: {{ .Values.service.port | quote }}
+        - name: WORLD_SIZE
+          value: {{ .Values.pytorch.distributed.worldSize | quote }}
+        - name: RANK
+          value: "0"
+        {{- end }}
+{{- if .Values.pytorch.extraEnvVars }}
+{{ toYaml .Values.pytorch.extraEnvVars | indent 8 }}
+{{- end }}
+        ports:
+        - name: pytorch
+          containerPort: {{ .Values.service.port }}
+        {{- if .Values.livenessProbe.enabled }}
+        livenessProbe:
+          exec:
+            command:
+            - python
+            - -c
+            - import torch; torch.__version__
+          initialDelaySeconds: {{ .Values.livenessProbe.initialDelaySeconds }}
+          periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
+          timeoutSeconds: {{ .Values.livenessProbe.timeoutSeconds }}
+          successThreshold: {{ .Values.livenessProbe.successThreshold }}
+          failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
+        {{- end }}
+        {{- if .Values.readinessProbe.enabled }}
+        readinessProbe:
+          exec:
+            command:
+            - python
+            - -c
+            - import torch; torch.__version__
+          initialDelaySeconds: {{ .Values.readinessProbe.initialDelaySeconds }}
+          periodSeconds: {{ .Values.readinessProbe.periodSeconds }}
+          timeoutSeconds: {{ .Values.readinessProbe.timeoutSeconds }}
+          successThreshold: {{ .Values.readinessProbe.successThreshold }}
+          failureThreshold: {{ .Values.readinessProbe.failureThreshold }}
+        {{- end }}
+        resources: {{ toYaml .Values.resources | nindent 12 }}
+        volumeMounts:
+        {{- if .Values.pytorch.configMap }}
+        - name: ext-files
+          mountPath: /app
+        {{- else if .Files.Glob "files/*" }}
+        - name: local-files
+          mountPath: /app
+        {{- else if .Values.pytorch.cloneFilesFromGit.enabled }}
+        - name: git-cloned-files
+          mountPath: /app
+        {{- end }}
+        - name: data
+          mountPath: {{ .Values.persistence.mountPath }}
+      volumes:
+      {{- if .Values.pytorch.configMap }}
+      - name: ext-files
+        configMap:
+          name: {{ .Values.pytorch.configMap }}
+      {{- else if .Files.Glob "files/*" }}
+      - name: local-files
+        configMap:
+          name: {{ include "pytorch.fullname" . }}-files
+      {{- else if .Values.pytorch.cloneFilesFromGit.enabled }}
+      - name: git-cloned-files
+        emptyDir: {}
+      {{- end }}
```

Comment:
Should we add warnings in case people set both `pytorch.configMap` and `pytorch.cloneFilesFromGit.enabled`?

Reply:
Not answered

Reply:
Sorry. This is documented in the README, but it may be good to add it.